File: 2016-08-15-WebViewExample/src/main/java/tech/thdev/webviewexample/view/main/presenter/MainContract.java
package tech.thdev.webviewexample.view.main.presenter;

import tech.thdev.kotlin_example_01.base.presenter.BasePresenter;
import tech.thdev.kotlin_example_01.base.presenter.BaseView;
import tech.thdev.webviewexample.webkit.listener.OnWebViewListener;

/**
 * Created by tae-hwan on 8/14/16.
 */

public interface MainContract {

    interface View extends BaseView<Presenter> {

        void loadUrl(String url);

        /**
         * WebView request url change
         */
        void setUrl(String url);

        /**
         * WebView scroll change
         * @param isTop If true, scroll to the top.
         */
        void webViewScrollChanged(boolean isTop);

        /**
         * WebView Progress changed
         */
        void webViewProgressChanged(int newProgress);
    }

    interface Presenter extends BasePresenter<View>, OnWebViewListener {

        void validationUrl(String text);

        /**
         * Default url load...
         */
        void defaultLoadUrl();
    }
}

File: app/src/main/java/tech/thdev/app/ui/main/presenter/MainContract.kt
package tech.thdev.app.ui.main.presenter

import tech.thdev.app.base.presenter.BasePresenter
import tech.thdev.app.base.presenter.BaseView
import tech.thdev.app.webkit.listener.OnWebViewListener

/**
 * Created by tae-hwan on 8/14/16.
 */
interface MainContract {
    interface View : BaseView<Presenter> {
        fun loadUrl(url: String)

        /**
         * WebView request url change
         */
        fun setUrl(url: String?)

        /**
         * WebView scroll change
         *
         * @param isTop If true, scroll to the top.
         */
        fun webViewScrollChanged(isTop: Boolean)

        /**
         * WebView Progress changed
         */
        fun webViewProgressChanged(newProgress: Int)
    }

    interface Presenter : BasePresenter<View>, OnWebViewListener {
        fun validationUrl(text: String?)

        /**
         * Default url load...
         */
        fun defaultLoadUrl()
    }
}